packages/devtools_app_shared/lib/src/service/connected_app.dart

```diff
@@ -50,8 +50,10 @@ class ConnectedApp {
   String? _operatingSystem;
 
   // TODO(kenz): investigate if we can use `libraryUriAvailableNow` instead.
-  Future<bool> get isFlutterApp async => _isFlutterApp ??=
-      await serviceManager!.libraryUriAvailable(flutterLibraryUri);
+  Future<bool> get isFlutterApp async =>
+      (_isFlutterApp ??=
+          await serviceManager?.libraryUriAvailable(flutterLibraryUri)) ??
```

Comment:
I don't think this change is necessary. `serviceManager` should never be null. If you did get a null exception my guess is that it was a bug due to hot restart